### PR TITLE
[FS-508] redfish dell bios config

### PR DESCRIPTION
**Pull request by diogomatsubara**
## What does this PR implement/change/remove?

Implement collection of bios configuration for Dell via redfish provider.

To try it out use this gist: https://gist.github.com/diogomatsubara/343043a6397514538d415e6e79f5b169 and set the env vars accordingly.

### Checklist
- [x] Tests added
- [x] Similar commits squashed

### The HW vendor this change applies to (if applicable)
Dell

## Description for changelog/release notes

```
Add functionality to get bios configuration from Dell systems using redfish
```

### CODE REVIEW

**Review by joelrebel — COMMENTED** [review]
Thanks for your work here! I've left a few suggestions.

File: internal/redfishwrapper/client.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ type Client struct {
 	user                  string
 	pass                  string
 	versionsNotCompatible []string // a slice of redfish versions to ignore as incompatible
-	client                *gofish.APIClient
+	Client                *gofish.APIClient
```
[the remark this answers is not in the document]
Comment (diogomatsubara) [reply internal/redfishwrapper/client.go]: Thanks for the pointer. I removed the commit that exported the client.

File: providers/redfish/bios.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package redfish
+
+import (
+	"context"
+
+	gofishrf "github.com/stmcginnis/gofish/redfish"
+)
+
+func (c *Conn) GetBiosConfiguration(ctx context.Context) (biosConfig map[string]string, err error) {
+	// TODO: replace uri parameter with configurable value
```
[the remark this answers is not in the document]
Comment (diogomatsubara) [reply providers/redfish/bios.go]: Indeed, much better this way. Thanks for the review! [621d498](https://github.com/bmc-toolbox/bmclib/pull/311/commits/621d498c7329996f7f5cad04035e2aff681a20d7) includes the changes suggested. Let me know what you think.

**Comment by diogomatsubara** [comment]
@joelrebel added the example in [209e24a](https://github.com/bmc-toolbox/bmclib/pull/311/commits/209e24ab49fd453f66b0f2eacc9a63724914c149)

**Review by joelrebel — APPROVED** [review]

**Pull request merged**